sklearn/ensemble/gradient_boosting.py

```diff
@@ -1417,11 +1420,21 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             # init state
             self._init_state()
 
-            # fit initial model - FIXME make sample_weight optional
-            self.init_.fit(X, y, sample_weight)
+            # fit initial model
+            try:
+                self.init_.fit(X, y, sample_weight=sample_weight)
+            except TypeError:
+                if sample_weight_was_none:
+                    self.init_.fit(X, y)
+                else:
+                    raise ValueError(
+                        "The initial estimator {} does not support sample "
+                        "weights yet.".format(self.init_.__class__.__name__))
```

Comment:
```suggestion
                        "weights.".format(self.init_.__class__.__name__))
```

Reply:
It was to point that it's not a definitive statement :)